### PR TITLE
Persist logs to disk + add network timeouts

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: hyperdash/server_manager.py [hunk 1]
```diff
@@ -173,13 +173,17 @@ def tick(self, sdk_run_uuid):
...
                 return False
 
-    def send_message(self, message, raise_exceptions=True):
+    def send_message(self, message, raise_exceptions=True, timeout_seconds=5):
         try:
             return self.s.post(
                 get_http_url(),
                 json=json.loads(message),
                 headers={AUTH_KEY_NAME: self.get_api_key()},
+                timeout=timeout_seconds
```
Comment (spence) [inline hyperdash/server_manager.py]: Trailing comma

File: hyperdash/hyper_dash.py [hunk 2]
```diff
@@ -189,24 +233,30 @@ def network_loop():
...
+                self.log_file.flush()
+
+                # Make a best-effort attempt to notify server that the run was
+                # canceled by the user, but don't wait for all messages to
+                # be flushed to server so we don't hang the user's terminal.
                 self.server_manager.send_message(
                     create_run_ended_message(self.current_sdk_run_uuid, "user_canceled"),
-                    raise_exceptions=False
+                    raise_exceptions=False,
+                    timeout_seconds=1
```
Comment (spence) [inline hyperdash/hyper_dash.py]: Trailing

File: setup.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from setuptools import setup
 
 
-version = "0.6.2"
+version = "0.7.2"
```
Comment (spence) [inline setup.py]: Shouldn't the patch version reset?

**Comment by spence** [comment]
Were we going to print the log file location once the run is complete?

**Comment by richardartoul** [comment]
@spence Yep, just added the print statement and addressed your other comments

**Pull request merged**